### PR TITLE
New tool: pysradb

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/pysradb/pysradb.xml [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+    #end if
+    #if $conditional_platform
+        #if $conditional_platform.platform == 'illumina'
+            #if $conditional_platform.layout
+                --layout $conditional_platform.layout
+            #end if
+        #end if
+    #end if
+    #if $conditional_platform
+        --platform $conditional_platform.platform
```
Comment (bgruening) [inline tools/pysradb/pysradb.xml:30]: are two words, like oxford nanopore, here really allowed? Do they need to be in quotes?

Comment (gallardoalba) [reply tools/pysradb/pysradb.xml]: It seems that works, but I included the quotes.

File: tools/pysradb/pysradb.xml [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+    #if 'graphs' in $output_options.output_files
+        --graphs 'daterange selection basecount'
+    #end if
+    --saveto 'output.tsv'
+    #if 'stats' in $output_options.output_files
+        > stats.txt
+        #if $output_options.verbosity == '2' or $output_options.verbosity == '3'
+            && echo '  Sequencing instrument:' >> stats.txt
+            #if $database == 'sra'
+                &&  awk -F '\t' '{print $11}' 'output.tsv' | tail -n +2 | sort | uniq -c | awk '{ print "    ", $0 }' >> stats.txt
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/pysradb/pysradb.xml]: Done, thanks!

File: tools/pysradb/pysradb.xml [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+                    <param name="organism" value="Homo sapiens"/>
+                    <param name="source" value="transcriptomic"/>
+                </conditional>
+                <section name="output_options">
+                    <param name="max" value="100"/>
+                    <param name="output_files" value="stats,graphs"/>
+                </section>
+                <output name="metadata_file" file="test_01.tabular" ftype="tsv"/>
+                <output name="stats" file="test_01_stats.txt" ftype="txt"/>
+                <output_collection name="graphs_collection">
```
Comment (bernt-matthias) [inline tools/pysradb/pysradb.xml:293]: Can you add the `count` attribute?

File: tools/pysradb/pysradb.xml [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+                    <validator type="regex">[0-9a-zA-Z]+</validator>
+                </param>
+                <param argument="--mbases" type="integer" min="0" value="" optional="true" 
+                    label="Sample size" help="Size of the sample rounded to the nearest megabase" />
+                <param argument="--publication-date" type="text" value="" optional="true" 
+                    label="Publication date" help="Publication date of the run in the format dd-mm-yyyy. 
+                        If a date range is desired, enter the start date, followed by end date, separated by a colon ':'.
+                        Example: 01-01-2010:31-12-2010">
+                    <sanitizer invalid_char="">
+                        <valid initial="string.letters,string.digits">
```
Comment (bernt-matthias) [inline tools/pysradb/pysradb.xml:244]: According to the help text `string.letters` are not needed, or?

File: tools/pysradb/pysradb.xml [hunk 5]
```diff
@@ -0,0 +1,520 @@
+<tool id='pysradb_search' name='pysradb search' version='@TOOL_VERSION@+galaxy@SUFFIX_VERSION@' profile='20.01'>
+    <description>query sequence metadata from SRA/ENA</description>
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/pysradb/pysradb.xml:2]: True, thanks!

File: tools/pysradb/pysradb.xml (hunk 4 above)
Comment (gallardoalba) [reply tools/pysradb/pysradb.xml:244]: Yes, you are right, thanks!

**Review by bgruening — APPROVED** [review]

**Pull request merged**